src/Agent.Sdk/Util/WhichUtil.cs

```diff
@@ -29,42 +29,71 @@ public static string Which(string command, bool require = false, ITraceWriter tr
-                    string pathExt = Environment.GetEnvironmentVariable("PATHEXT");
-                    if (string.IsNullOrEmpty(pathExt))
-                    {
-                        // XP's system default value for PATHEXT system variable
-                        pathExt = ".com;.exe;.bat;.cmd;.vbs;.vbe;.js;.jse;.wsf;.wsh";
-                    }
-
-                    string[] pathExtSegments = pathExt.Split(new string[] { ";" }, StringSplitOptions.RemoveEmptyEntries);
 
-                    // if command already has an extension.
-                    if (pathExtSegments.Any(ext => command.EndsWith(ext, StringComparison.OrdinalIgnoreCase)))
+                    if (PlatformUtil.RunningOnOS == PlatformUtil.OS.Windows)
                     {
-                        try
+                        string pathExt = Environment.GetEnvironmentVariable("PATHEXT");
+                        if (string.IsNullOrEmpty(pathExt))
                         {
-                            matches = Directory.GetFiles(pathSegment, command);
+                            // XP's system default value for PATHEXT system variable
+                            pathExt = ".com;.exe;.bat;.cmd;.vbs;.vbe;.js;.jse;.wsf;.wsh";
                         }
-                        catch (UnauthorizedAccessException ex)
+
+                        string[] pathExtSegments = pathExt.Split(new string[] { ";" }, StringSplitOptions.RemoveEmptyEntries);
+
+                        // if command already has an extension.
+                        if (pathExtSegments.Any(ext => command.EndsWith(ext, StringComparison.OrdinalIgnoreCase)))
                         {
-                            trace?.Info("Ignore UnauthorizedAccess exception during Which.");
-                            trace?.Verbose(ex.ToString());
-                        }
+                            try
+                            {
+                                matches = Directory.GetFiles(pathSegment, command);
+                            }
+                            catch (UnauthorizedAccessException ex)
+                            {
+                                trace?.Info("Ignore UnauthorizedAccess exception during Which.");
+                                trace?.Verbose(ex.ToString());
+                            }
 
-                        if (matches != null && matches.Length > 0)
+                            if (matches != null && matches.Length > 0)
+                            {
+                                trace?.Info($"Location: '{matches.First()}'");
+                                return matches.First();
+                            }
+                        }
+                        else
                         {
-                            trace?.Info($"Location: '{matches.First()}'");
-                            return matches.First();
+                            string searchPattern;
+                            searchPattern = StringUtil.Format($"{command}.*");
```

Comment:
why separate lines?

Reply:
Dunno, I tried not to touch anything. But I agree, this is gross.